### PR TITLE
Start and stop recording

### LINKED ISSUE #103: Start and Stop Recording
**What and why do you want to change?**
Right now recording data runs all the time which can cause problems for memory slowdown see (issue #88 ) and as we approach a testing dashboard phase it would be nice to stop and start recording.

### PR DESCRIPTION
**Pull request by rednazer**
# Description

- Please include a summary of the change and which issue (if applicable) is fixed
I added the feature to start and stop the recording of data and changed naming convention of exported json files to include seconds (this was to fix a bug where recording 2 times in the same minute would overwrite the first data record).

- Resolves #103  (delete if n/a)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

# Checklist

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (readme)
- [X] My changes generate no new warnings or errors
- [X] I have added tests/examples that prove my fix is effective or that my feature works
- [X] New and existing lint guidelines/tests/examples pass with my changes (run npm test locally)

### CODE REVIEW

File: src/public/javascripts/datainterfacing.js [hunk 1]
```diff
@@ -6,19 +6,55 @@ const events = require('events');
...
 const packetHandler = new events.EventEmitter();
 module.exports.packetHandler = packetHandler;
 
+// Data recording variable
+module.exports.isDataRecording = false;
+// Changes the data recording variable from the handler
+const recordingEvent = new events.EventEmitter();
+module.exports.recordingEvent = recordingEvent;
+// Sets that data is recording
+recordingEvent.on(true, () => {
```
Comment (EUdds) [inline src/public/javascripts/datainterfacing.js]: Event names should be strings to avoid confusion

File: src/index.html [hunk 2]
```diff
@@ -146,6 +146,9 @@ <h5>Clear Charts</h5>
           <button id="brakingFault" class="stateButtonInactive">
             <p id="pbrakeFault">Not in Use</p>
           </button>
```
Comment (EUdds) [inline src/index.html]: We should color these buttons

Comment (EUdds) [reply src/index.html]: @LukeHouge Can you take care of this?

Comment (EUdds) [reply src/index.html]: Jk I got it

**Comment by EUdds** [comment]
Overall this looks really awesome!! Just adjust the event names and it'll be ready for master!!

**Comment by rednazer** [comment]
I fixed the naming convention, added a comment on createCache and removed the removed element in dynamicloading.js.

**Comment by EUdds** [comment]
Lol https://www.reddit.com/r/javascript/comments/66ij5p/what_is_an_eslint_rule_that_you_could_not_do/dgiwl5i?utm_source=share&utm_medium=web2x

**Pull request merged**